tests/providers/google/marketing_platform/hooks/test_analytics.py

```diff
@@ -0,0 +1,75 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import unittest
+from unittest import mock
+
+from airflow.providers.google.marketing_platform.hooks.analytics import GoogleAnalyticsListAccountsHook
+from tests.providers.google.cloud.utils.base_gcp_mock import mock_base_gcp_hook_default_project_id
+
+API_VERSION = "v3"
+GCP_CONN_ID = "google_cloud_default"
+
+
+class TestGoogleAnalyticsHook(unittest.TestCase):
+    def setUp(self):
+        with mock.patch(
+            "airflow.providers.google.cloud.hooks.base.CloudBaseHook.__init__",
+            new=mock_base_gcp_hook_default_project_id,
+        ):
+            self.hook = GoogleAnalyticsListAccountsHook(API_VERSION, GCP_CONN_ID)
+
+    @mock.patch(
+        "airflow.providers.google.marketing_platform.hooks."
+        "analytics.GoogleAnalyticsListAccountsHook._authorize"
+    )
+    @mock.patch(
+        "airflow.providers.google.marketing_platform.hooks.analytics.build"
+    )
+    def test_gen_conn(self, mock_build, mock_authorize):
+        result = self.hook.get_conn()
+        mock_build.assert_called_once_with(
+            "analytics",
+            API_VERSION,
+            http=mock_authorize.return_value,
+            cache_discovery=False,
+        )
+        self.assertEqual(mock_build.return_value, result)
+
+    @mock.patch(
+        "airflow.providers.google.marketing_platform.hooks."
+        "analytics.GoogleAnalyticsListAccountsHook.get_conn"
+    )
+    def test_list_accounts(self, get_conn_mock):
```

Comment:
should I add tests for checking returned type?

Reply:
Yes. You should.

Reply:
It's not needed, because you use mock. It will always be a mock object.

Reply:
Ok.